src/topology/sheaves/locally_surjective.lean

```diff
@@ -0,0 +1,162 @@
+
+We prove that these are equivalent.
+
+-/
+
+universes v u
+
+noncomputable theory
+
+open category_theory
+open topological_space
+open opposite
+
+section locally_surjective
+
+/-- Let C be a concrete category, X a topological space. -/
+variables {C : Type u} [category.{v} C] [concrete_category.{v} C] {X : Top.{v}}
+
+/-- Let ℱ, 𝒢 : (opens X)ᵒᵖ ⥤ C be C-valued presheaves on X. -/
+variables {ℱ : X.presheaf C} {𝒢 : X.presheaf C}
+
+/-- When U is an open set, we introduce the notation "Γ_ ℱ U"
+for the set of sections of ℱ over U.
+-/
+@[nolint has_inhabited_instance] -- This may be empty.
+def sections_of_presheaf_on_open (ℱ : X.presheaf C) (U : opens X) : Type v :=
+  (forget C).obj (ℱ.obj (op U))
+local notation `Γ_` : 80 := sections_of_presheaf_on_open
+
+/-- When i : V ⟶ U is an inclusion of an open set V into an open set U,
+and s ∈ Γ_ ℱ U, we write "s|_i" for the restriction of s to V. -/
+def restrict_along {ℱ : X.presheaf C} {U : opens X} {V : opens X}
+  (s : Γ_ ℱ U) (i : V ⟶ U) : Γ_ ℱ V := (forget C).map (ℱ.map i.op) s
+local infix `|_` : 80 := restrict_along
+
+/-- When T : ℱ ⟶ 𝒢 is a natural transformation, and s ∈ Γ_ ℱ U, we
+write "T _* s" for the image of s under the map T_U : Γ_ ℱ U ⟶ Γ_ 𝒢 U. -/
+def map_on_sections {U : opens X} (T : ℱ ⟶ 𝒢) (s : Γ_ ℱ U) :
+  Γ_ 𝒢 U := (forget C).map (T.app (op U)) s
+local infix ` _* ` : 80 := map_on_sections
+
+/-- A map of presheaves T : ℱ ⟶ 𝒢 is **locally surjective** if for
+any open set U, section t over U, and x ∈ U, there exists an open set
+x ∈ V ⊆ U such that $T_*(s_V) = t|_V$. -/
+def is_locally_surjective (T : ℱ ⟶ 𝒢) :=
+  ∀ (U : opens X) (t : Γ_ 𝒢 U) (x : X) (hx : x ∈ U),
+  ∃ (V : opens X) (ι : V ⟶ U) (hxV : x ∈ V) (s : Γ_ ℱ V),
+  T _* s = t |_ ι
+
+section surjective_on_stalks
+
+variables [category_theory.limits.has_colimits C]
+
+/-- When (x : X) is a point, we introduce the notation "Γₛₜ ℱ x" for
+the (underlying object of) the stalk of ℱ at x. -/
+@[nolint has_inhabited_instance] -- This may be empty.
+def stalk_set (ℱ : X.presheaf C) (x : X) :=
+  (forget C).obj (ℱ.stalk x)
+local notation `Γₛₜ` : 80 := stalk_set
+
+/-- When (T : ℱ ⟶ 𝒢) is a map of presheaves, we introduce the notation "T _ₛₜ x"
+for the induced map of (underlying objects of) stalks. -/
+def map_on_stalks (T : ℱ ⟶ 𝒢) (x : X) :
+  Γₛₜ ℱ x ⟶ Γₛₜ 𝒢 x := (forget C).map ((Top.presheaf.stalk_functor C x).map T)
+local infix `_ₛₜ` : 80 := map_on_stalks
+
+/-- An equivalent condition for a map of presheaves to be locally surjective
+is for all the induced maps on stalks to be surjective. -/
+def is_surjective_on_stalks (T : ℱ ⟶ 𝒢) :=
+  ∀ (x : X), function.surjective (T _ₛₜ x)
+
+variables [category_theory.limits.preserves_filtered_colimits (forget C)]
+
+/-- Being locally surjective is equivalent to being surjective on stalks. -/
+lemma locally_surjective_iff_surjective_on_stalks (T : ℱ ⟶ 𝒢) :
+  is_locally_surjective T ↔ is_surjective_on_stalks T :=
```

Comment:
For example, I think the `is_surjective_on_stalks` definition isn't really nessecary, and practically it will probably be unfolded whenever it is met.
Under the namespace and the local instances, the type signature could become
```suggestion
  is_locally_surjective T ↔ ∀ x, function.surjective ((stalk_functor C x).map T) :=
```
Which is just as concise and clear IMHO.